Fix for tooltip blinking

captainwasabi opened the pull request:
Moved where active tooltips are canceled to wait until the mouse actually moves off the control.
Closes #3835.

captainwasabi committed:
Moved _gui_cancel_tooltip(); to check if the gadget has changed, rather than just the mouse moving.

captainwasabi committed:
moved _guy_cancel_tooltip to only cancel if the mouse moves off the control.

neikeq commented:
Some notes:
- There seems to be a problem with your git config (check the e-mail). Your commits are not linked to your Github profile. This is not a problem for merging, but the commits won't be attributed to your profile in Github's contributors page.
- If this PR definitely closes all the mentioned issues, you can add one of [these keywords](https://help.github.com/articles/closing-issues-via-commit-messages/#keywords-for-closing-issues) before the issue id (in the PR description) and they will be closed automatically when this PR gets merged. e.g: Closes #4686

captainwasabi commented:
Ah, I used a different address setting up git than what I use to log in here.  Added it to my profile, correctly attributed now.

Added closed keyword

Thanks!  TIL
